bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/processor/ArticleProcessor.java

```diff
@@ -0,0 +1,270 @@
+		getRecentPatientPrescriptions(actEncounter.getPatient()).stream().filter(prescription -> {
+			Object extInfo = prescription.getExtInfo(ch.elexis.core.model.prescription.Constants.FLD_EXT_VERRECHNET_ID);
+			return extInfo != null && billed.getId().equals(extInfo.toString());
+		}).findFirst().ifPresent(prescription -> {
+			prescription.setDosageInstruction(signature.getSignatureAsDosisString());
+			prescription.setRemark(signature.getComment());
+			prescription.setDateFrom(billed.getEncounter().getDate().atStartOfDay());
+			CoreModelServiceHolder.get().save(prescription);
+		});
+	}
+
+	private boolean fixOrReserveWithArticleExists(List<IPrescription> prescriptions, IArticle selectedArticle) {
+		return prescriptions.stream()
+				.anyMatch(prescription -> prescription.getArticle().getId().equals(selectedArticle.getId())
+						&& (prescription.getEntryType() == EntryType.FIXED_MEDICATION
+								|| prescription.getEntryType() == EntryType.RESERVE_MEDICATION));
+	}
+
+	public void updatePrescriptionsWithDosage(IBilled billed) {
+	    List<IPrescription> prescriptions = getRecentPatientPrescriptions(actEncounter.getPatient());
+	    List<IPrescription> reversedPrescriptions = new ArrayList<>(prescriptions);
+	    Collections.reverse(reversedPrescriptions);
+	    Optional<IPrescription> lastPrescriptionWithDosage = reversedPrescriptions.stream()
+	    	    .filter(prescription -> isMatchingPrescription(prescription, billed))
+	    	    .findFirst();
+	    lastPrescriptionWithDosage.ifPresent(lastPrescription -> prescriptions.stream()
+	        .filter(prescription -> {
+	            IArticle article = prescription.getArticle();
+	            return article != null && article.getGtin() != null && billed.getCode().equals(article.getGtin());
+	        })
+	        .filter(prescription -> {
+	            String dosageInstruction = prescription.getDosageInstruction();
+	            return dosageInstruction == null || dosageInstruction.isEmpty()
+	                    || !dosageInstruction.equals(lastPrescription.getDosageInstruction());
+	        })
+	        .forEach(prescription -> {
+	            prescription.setDosageInstruction(lastPrescription.getDosageInstruction());
+	            prescription.setRemark(lastPrescription.getRemark());
+	            prescription.setDateFrom(actEncounter.getDate().atStartOfDay());
+	            CoreModelServiceHolder.get().save(prescription);
+	        }));
+	}
+
+	private boolean isMatchingPrescription(IPrescription prescription, IBilled billed) {
+	    IArticle article = prescription.getArticle();
+	    String dosage = prescription.getDosageInstruction();
+	    return article != null && article.getGtin() != null && billed.getCode().equals(article.getGtin())
+	            && dosage != null && !dosage.isEmpty();
+	}
+
+	public static List<IPrescription> getRecentPatientPrescriptions(IPatient patient) {
+		return ContextServiceHolder.get().getTyped(IEncounter.class).map(IEncounter::getPatient)
```

Comment:
Der `IPatient` kommt hier als Übergabeparameter in die Methode, dann wird vom `ContextService` der aktive `IEncounter` geholt, und von dem der `IPatient` genommen. Bitte entweder den Übergabeparameter entfernen, oder den übergebenen `IPatient` verwenden.